django/db/backends/postgresql/operations.py

```diff
@@ -271,5 +274,18 @@ def explain_query_prefix(self, format=None, **options):
             prefix += ' (%s)' % ', '.join('%s %s' % i for i in extra.items())
         return prefix
 
-    def ignore_conflicts_suffix_sql(self, ignore_conflicts=None):
-        return 'ON CONFLICT DO NOTHING' if ignore_conflicts else super().ignore_conflicts_suffix_sql(ignore_conflicts)
+    def conflicts_suffix_sql(
+        self, opts, fields, on_conflicts=ON_CONFLICTS_NONE, update_fields=[], unique_fields=[]
+    ):
+        if on_conflicts == ON_CONFLICTS_IGNORE:
+            return 'ON CONFLICT DO NOTHING'
+
+        elif on_conflicts == ON_CONFLICTS_UPDATE:
+            result = 'ON CONFLICT(%s) DO UPDATE SET ' % (', '.join(unique_fields))
+            result += ', '.join('%s=excluded.%s' % (field, field) for field in update_fields)
+            return result
+
+        return super().conflicts_suffix_sql(
+            opts, fields, on_conflicts=on_conflicts, update_fields=update_fields,
+            unique_fields=unique_fields
+        )
```

Comment:
```suggestion
    def on_conflict_suffix_sql(self, opts, fields, on_conflict=None, update_fields=None, unique_fields=None):
        if on_conflict == OnConflict.IGNORE:
            return 'ON CONFLICT DO NOTHING'
        if on_conflict == OnConflict.UPDATE:
            return 'ON CONFLICT(%s) DO UPDATE SET %s' % (
                ', '.join(unique_fields or ()),
                ', '.join(f'{field}=excluded.{field}' for field in update_fields or ()),
            )
        return super().on_conflict_suffix_sql(opts, fields, on_conflict, update_fields, unique_fields)
```

If we have `unique_fields` in the signature here, then it should also be in the signature of the super class, not falling back to `**kwargs`.

As mentioned in https://github.com/django/django/pull/13065#discussion_r668635778 we should probably be quoting identifiers here properly too.